AI #1311: Add Contract columns to Cost and Usage dataset (#1327)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/columns/contractid.md

# Contract ID

> A Contract ID is a provider-assigned identifier for a contract describing the agreed terms between a provider and a customer.  Contracts often describe one or more [*negotiated discounts*](#glossary:negotiated-discount). Contracts can include commitment to a certain amount of spend or usage over an agreed period of time.

Reviewer: ```suggestion
A Contract ID is a provider-assigned identifier for a contract describing the agreed terms between a provider and a customer. Contracts often describe one or more [*negotiated discounts*](#glossary:negotiated-discount). Contracts can include time-bound spend or usage commitments.
```

> A Contract ID is a provider-assigned identifier for a contract describing the agreed terms of one or more [*negotiated discounts*](#glossary:negotiated-discount) between a provider and a customer. Such discounts are typically the result of a customer committing to a certain amount of spend or usage over an agreed period of time.

Reviewer 2: Contracts do not require negotiated discounts. This implies that contracts must include negotiated discounts, when that is not the case.

Author: Fair point.  I would love a suggestion, if you have it!  How about this:

```suggestion
A Contract ID is a provider-assigned identifier for a contract describing the agreed terms between a provider and a customer.  Contracts often describe one or more [*negotiated discounts*](#glossary:negotiated-discount), obtained through the customer's commitment to a certain amount of spend or usage over an agreed period of time.
```

Reviewer 2: Minor change as contracts can include neither spend or usage commits.  You could go even further and remove the spend or usage commit entirely. 

```suggestion
A Contract ID is a provider-assigned identifier for a contract describing the agreed terms between a provider and a customer.  Contracts often describe one or more [*negotiated discounts*](#glossary:negotiated-discount). Contracts can include commitment to a certain amount of spend or usage over an agreed period of time.
```

Author: I'd like to keep that reference because, while you are correct that they may not include such commitments, they represent the bulk of what we're considering here. 

Accepting your suggestion!

The ContractId column adheres to the following requirements:

> * ContractId MUST be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset) when the provider supports *negotiated discounts*.

Author: We may need to revisit the conditional nature of this requirement, per discussion in Aug 12 TF1.

@marc-perreaut and @rileyjenk will produce suggestions.

Author: @SatyaVandrangi1 is concerned that if we make this more globally applicable in terms of presence, then we are not necessarily facilitating just an MVP of simple drawdown contracts in 1.3.

Reviewer 3: I would tend to think the Contract ID should be always present, as there should be always a contract when charging cost.
Indeed, the Contracted Cost column must be always present.

Reviewer 4: I would agree there's always a contract, but that is separate from the commitment, which I believe is what this is trying to track. I don't think we can merge those concepts.
